corehq/middleware.py

```diff
@@ -143,19 +143,27 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             return
 
         secure_session = request.session.get('secure_session')
-        timeout = settings.SECURE_TIMEOUT if secure_session else settings.INACTIVITY_TIMEOUT
         domain = getattr(request, "domain", None)
+        domain_obj = Domain.get_by_name(domain) if domain else None
         now = datetime.datetime.utcnow()
 
         # figure out if we want to switch to secure_sessions
         change_to_secure_session = (
             not secure_session
             and (
-                (domain and Domain.is_secure_session_required(domain))
+                (domain_obj and domain_obj.secure_sessions)
                 or self._user_requires_secure_session(request.couch_user)))
 
+        timeout = None
+        if secure_session or change_to_secure_session:
+            if domain_obj:
+                timeout = domain_obj.secure_timeout
+            if not timeout:
+                timeout = settings.SECURE_TIMEOUT
```

Comment:
any reason not to go with the one-line version here? i guess cause the null `domain_obj` check makes it more complicated?

Reply:
yeah, it's not beautiful code either way, going to leave this as-is